R/UK.R

```diff
@@ -496,6 +496,52 @@ UK <- R6::R6Class("UK",
           release_date = self$release_date
         )
       return(clean_data)
+    },
+
+    #' @description Specific tests for UK data. In addition to generic tests ran
+    #' by `DataClass$test()` data for NHS regions are downloaded and ran through
+    #' the same generic checks (test_cleaning, test_processing, test_return). If
+    #' download = TRUE or a snapshot file is not found, the nhs data is
+    #' downloaded and saved to the snapshot location provided. If an existing
+    #' snapshot file is found then this data is used in the next tests.
+    #' Tests data can be downloaded, cleaned, processed and returned.
+    #' @param self_copy R6class the object to test
+    #' @param download logical. To download the data (TRUE) or use a snapshot
+    #' (FALSE). Defaults to FALSE.
+    #' @param nhs_included_path character_array the path to save the downloaded
+    #' nhs data snapshot to.
+    specific_tests = function(self_copy, download = FALSE,
+                              nhs_included_path) {
+      if (missing(nhs_included_path)) {
+        message_verbose(
+          verbose = self_copy$verbose,
+          "nhs_included_path not provided, skipping test."
+        )
+        return(invisible(NULL))
+      }
+      if (self_copy$level == "1") {
+        self_copy$data_name <- "UK level 1 with 'nhsregions=TRUE'"
+        self_copy$nhsregions <- TRUE
```

Comment:
I would have thought these bits would only be triggered if test all was true of if the nhsregions was true?